src/orchard/java/resource.clj

```diff
@@ -0,0 +1,52 @@
+(ns orchard.java.resource
+  "Resolve JVM resource-related paths."
+  {:added "0.5.0"}
+  (:require
+   [clojure.java.io :as io]
+   [orchard.classpath :as cp]
+   [orchard.misc :as u]))
+
+(defn- trim-leading-separator
```

Comment:
It'd be nice if this function (and all the others) had some docstrings, even though they are relatively simple.